cmd/karlsenminer/config.go

```diff
@@ -83,7 +83,8 @@ func parseConfig() (*configFlags, error) {
 	}
 
 	if cfg.MiningAddr == "" {
-		return nil, errors.New("--miningaddr is required")
+		fmt.Fprintln(os.Stderr, errors.New("Error parsing command-line arguments: --miningaddr is required"))
+		os.Exit(42)
```

Comment:
I will change it to `os.Exit(1)` as it is used in every other place to exit with non-zero status.